e2e/web/src/web.test.ts

```diff
@@ -188,8 +187,50 @@ describe('Web Components Applications', () => {
     ).not.toThrow();
   }, 1000000);
 
-  it('should run default jest tests', async () => {
-    await expectJestTestsToPass('@nrwl/web:app');
+  it('should support custom webpackConfig option', async () => {
+    const appName = uniq('app');
+    runCLI(`generate @nrwl/web:app ${appName} --no-interactive`);
+
+    updateProjectConfig(appName, (config) => {
+      config.targets.build.options.webpackConfig = `apps/${appName}/webpack.config.js`;
+      return config;
+    });
+
+    // Return sync function
+    updateFile(
+      `apps/${appName}/webpack.config.js`,
+      `
+      module.exports = (config, context) => {
+        return config;
+      };
+    `
+    );
+    runCLI(`build ${appName} --outputHashing none`);
+    checkFilesExist(`dist/apps/${appName}/main.esm.js`);
+
+    // Return async function
+    updateFile(
+      `apps/${appName}/webpack.config.js`,
+      `
+      module.exports = async (config, context) => {
+        return config;
+      };
+    `
+    );
+    runCLI(`build ${appName} --outputHashing none`);
+    checkFilesExist(`dist/apps/${appName}/main.esm.js`);
```

Comment:
Should we add a statement to clear out the dist in between the last section and this? Currently if something was wrong with the build and it failed to clear the test would still pass.